client/src/document/toolbar/flaws.tsx

```diff
@@ -15,7 +14,7 @@ interface FlatFlaw {
 }
 
 const FLAWS_HASH = "#_flaws";
-export function ToggleDocumentFlaws({ flaws }: Pick<Doc, "flaws">) {
+export function ToggleDocumentFlaws({ doc }: { doc: Doc }) {
```

Comment:
Let's nerd out about this a little. I doubt there's a right and wrong. 
I like drop-drilling over context passing in general because it feels more visually available and apparent. 
But we all have a tendency to write code like this:
```javascript
const bigData = { ... }
<GrandDad data={bigData.data}>
  /* In GrandData */
  <Dad stuff={data.stuff}>
     /* In Dad */
    <Teenager detail={stuff.detail}>
      /* In Teenager */
      <Child value={detail.key}>
```

And then some day comes when reality hits the fan and you realize that the `Child` component actually needs something from that `bigData` at the top of the "prop passing". And now you have to rewrite every signature of every component along the way. I'm not talking about using `{...props}` necessarily, but the alternative would be:

```javascript
const bigData = { ... }
<GrandDad bigData={bigData}>
  /* In GrandData */
  <Dad bigData={bigData}>
     /* In Dad */
    <Teenager bigData={bigData}>
      /* In Teenager */
      <Child bigData={bigData}>
```
What's the harm? Suppose that `<Teenager>` only depends on a small portion of `bigData` so you might worry about unnecessary renderings. But does that even happen? First of all, you have the virtual DOM and second of all, if a parent component needs to re-render, it's going to trigger a re-render to children anyway. 

I can see pros and cons with these two opposing patterns and pros and cons with contexts. Just eager to hear what you feel @Gregoor  on this pattern topic.

Reply:
I'd err on the side of rather passing less than more data in. Reasons being:
- it reduces "apparent" coupling. Using shared data adds uncertainty to every change of its schema, as there are more places where it _could_ break
- it also leads to more monolithic data structures which makes for less reusable code. In your scenario shit hits the fan because you need more data. Another scenario to consider is you wanting to reuse a component in a context where a subset of bigData is available but not in the same form, or split up differently. Re-shaping it into the component's specific needs is bound to be easier than to rebuild the monolith
- the method by which I'd reduce prop-drilling is either considering context or, if possible, moving Child rendering up in the hierarchy. e.g.:
```jsx
const GrandDad = ({children}) => <Dad><Teenager>{children}</Teenager></Dad>;

const Root = ({bigData}) => <GrandDad><Child key={bigData.key}/></GrandDad>;
```

Reply:
That said, it's always a balancing act and trying to anticipate likelihoods of different kinds of changes, so I generally agree that there's no clear right or wrong.

Reply:
I'm pondering what's best for this use case and coming up a bit empty atm. I'd have probably passed filepath in, instead of the whole doc. But I think we're deep enough in the waters of ambiguity where no path is obviously more attractive (at least to me) that both options seem fine.

Reply:
Why is it so hard to make a "decision" on this? 
I think I'll stick to what I have in this PR here. I can definitely smell that in the future we're going to want more stuff out of that `doc` rather than picking out specifically, and only, want the `<BrokenLinks>` component needs. 

Well, let's never give up and keep questioning ourselves all the time. Sad, but most sustainable in the long run.